includes/admin/class-wc-admin-addons.php

```diff
@@ -534,6 +534,69 @@ class="addons-img"
 		<?php
 	}
 
+	/**
+	 * Handles the output of a full-width block.
+	 *
+	 * @param array $section Section data.
+	 */
+	public static function output_promotion_block( $section ) {
+		if (
+			! current_user_can( 'install_plugins' ) ||
+			! current_user_can( 'activate_plugins' )
+		) {
+			return;
+		}
+
+		$section_object = (object) $section;
+
+		if ( ! empty( $section_object->geo_allow_list ) ) {
```

Comment:
should not we do the same for block list as well? as of now blocklist won't won't work

Reply:
Good point – added in 7202f19631e474d1adc93d83acc1499bab0b6b3d.